arch/xtensa/src/common/xtensa_schedsigaction.c

```diff
@@ -155,6 +156,9 @@ void up_schedule_sigaction(struct tcb_s *tcb, sig_deliver_t sigdeliver)
                   (PS_INTLEVEL(XCHAL_EXCM_LEVEL) | PS_UM |
                    PS_WOE | PS_CALLINC(1));
 #endif
+#ifndef CONFIG_BUILD_FLAT
+              mpu_raiseprivilege(CURRENT_REGS);
```

Comment:
should we use a general prefix other than mpu? I am afraid that other chip may implement the protection by the different hardware module.

Reply:
What would be your suggestion? I had spent some time thinking, but prefixing it with `mpu_` was the one I considered most suitable.

Reply:
xtensa_? MPU is normally used to change the memory region protection attribute, low/raise cpu/dsp priority is normally done by core self.

Reply:
I preferred to not prefix it with `xtensa_` to avoid confusion and also to prevent any kind of clash with the Xtensa MPU block, which ESP32 does not embed.

Newer Xtensa-based Espressif chips (ESP32-S2 and ESP32-S3) contain the PMS peripheral, which also performs this hardware and software resource management.

Reply:
> low/raise cpu/dsp priority is normally done by core self

How about prefixing it with `up_` and declaring it on `include/nuttx/arch.h`?
Then, at architectural level, there could exist a weak implementation of those functions which, if needed, could be overriden by chip-specific implementation (the case for Espressif Xtensa-based chips).
The RISC-V implementation manipulates the `STATUS_PPP` flag and could also benefit from this abstraction.

Reply:
> > low/raise cpu/dsp priority is normally done by core self
> 
> How about prefixing it with `up_` and declaring it on `include/nuttx/arch.h`?

but @patacongo prefer arch specific interface use arm_, riscv_ or xtensa_ prefix, and the common interface shared by all arch use up_ prefix.

> Then, at architectural level, there could exist weak implementation of those functions which, if needed, could be overriden by chip-specific ones (the case for Espressif chips).

Weak symbol isn't work very well with the static library sometime:
https://stackoverflow.com/questions/13089166/how-to-make-gcc-link-strong-symbol-in-static-library-to-overwrite-weak-symbol

> The RISC-V implementation manipulates the `STATUS_PPP` flag and could also benefit from this abstraction.

Reply:
> but @patacongo prefer arch specific interface use arm_, riscv_ or xtensa_ prefix, and the common interface shared by all arch use up_ prefix.

I agree, my suggestion was bad, after all, those symbols are only used within the architecture implementation so they shouldn't be present on the interface `arch.h`.

> Weak symbol isn't work very well with the static library sometime:
https://stackoverflow.com/questions/13089166/how-to-make-gcc-link-strong-symbol-in-static-library-to-overwrite-weak-symbol

This shouldn't be a problem to NuttX build system, since all the libraries are passed to the linker within a group (`--start-group`/`--end-group`).

Reply:
@xiaoxiang781216 I've renamed the `mpu_` prefix to `xtensa_`, PTAL.

Also, I've added some information about the `knsh` defconfig into the Documentation page.